cmake/extensions.cmake

```diff
@@ -128,69 +128,186 @@ endfunction()
+
+function(zephyr_get_system_include_directories_for_lang lang i)
+  get_property_and_add_prefix(flags zephyr_interface INTERFACE_SYSTEM_INCLUDE_DIRECTORIES -isystem)
+
+  process_flags(${lang} flags output_list)
+
+  set(${i} ${output_list} PARENT_SCOPE)
+endfunction()
+
+function(zephyr_get_compile_definitions_for_lang lang i)
+  get_property_and_add_prefix(flags zephyr_interface INTERFACE_COMPILE_DEFINITIONS -D)
+
+  process_flags(${lang} flags output_list)
+
+  set(${i} ${output_list} PARENT_SCOPE)
+endfunction()
+
+function(zephyr_get_compile_options_for_lang lang i)
+  get_property(flags TARGET zephyr_interface PROPERTY INTERFACE_COMPILE_OPTIONS)
+
+  process_flags(${lang} flags output_list)
+
+  set(${i} ${output_list} PARENT_SCOPE)
+endfunction()
+
+function(process_flags lang input output)
+  # The flags might contains compile language generator expressions that
+  # look like this:
+  # $<$<COMPILE_LANGUAGE:CXX>:-fno-exceptions>
+  #
+  # Flags that don't specify a language like this apply to all
+  # languages.
+  #
+  # See COMPILE_LANGUAGE in
+  # https://cmake.org/cmake/help/v3.3/manual/cmake-generator-expressions.7.html
+  #
+  # To deal with this, we apply a regex to extract the flag and also
+  # to find out if the language matches.
+  #
+  # If this doesn't work out we might need to ban the use of
+  # COMPILE_LANGUAGE and instead partition C, CXX, and ASM into
+  # different libraries
+  set(languages C CXX ASM)
+
+  set(tmp_list "")
+
+  foreach(flag ${${input}})
+    set(is_compile_lang_generator_expression 0)
+    foreach(l ${languages})
+      if(flag MATCHES "<COMPILE_LANGUAGE:${l}>:([^>]+)>")
+        set(is_compile_lang_generator_expression 1)
+        if(${l} STREQUAL ${lang})
+          list(APPEND tmp_list ${CMAKE_MATCH_1})
+          break()
+        endif()
+      endif()
+    endforeach()
+
+    if(NOT is_compile_lang_generator_expression)
+      list(APPEND tmp_list ${flag})
+    endif()
+  endforeach()
+
+  set(${output} ${tmp_list} PARENT_SCOPE)
+endfunction()
+
+function(convert_list_of_flags_to_string_of_flags ptr_list_of_flags string_of_flags)
+  # Convert the list to a string so we can do string replace
+  # operations on it and replace the ";" list separators with a
+  # whitespace so the flags are spaced out
+  string(REPLACE ";"  " "  locally_scoped_string_of_flags "${${ptr_list_of_flags}}")
+
+  # Set the output variable in the parent scope
+  set(${string_of_flags} ${locally_scoped_string_of_flags} PARENT_SCOPE)
+endfunction()
+
 macro(get_property_and_add_prefix result target property prefix)
   get_property(target_property TARGET ${target} PROPERTY ${property})
   foreach(x ${target_property})
     list(APPEND ${result} ${prefix}${x})
   endforeach()
 endmacro()
 
+# The below macros are deprecated in favour of
+# zephyr_get_${type}_for_lang[_as_string](lang x), they will be
+# removed in a future release.
+####################### BEGIN DEPRECATED ###################################################
```

Comment:
why not remove them now? There were never released, so they should just be dropped in favor of the above.

Reply:
Done.